### PR TITLE
Restore Pre-1.8 Server Spawn Protection Behavior closes #15

### LINKED ISSUE #15: Restore Pre-1.8 Server Spawn Protection Behavior
As some may know, in 1.8 the server disables spawn point protection if the server has no known OPs. In my opinion, this is an incredibly stupid design decision and, as evidenced by looking at various forum posts, many server admins were under a false impression that spawn protection broke.

Put simply, you shouldn't HAVE to have an OP specified for spawn protection work. Its an easy change to DedicatedServer to remove the if check.

@SpongePowered/developers thoughts?

### PR DESCRIPTION
**Pull request by ryantheleach**
@Zidane commented on 29 Apr https://github.com/SpongePowered/SpongeCommon/issues/15
>As some may know, in 1.8 the server disables spawn point protection if the server has no known OPs. In my opinion, this is an incredibly stupid design decision and, as evidenced by looking at various forum posts, many server admins were under a false impression that spawn protection broke.
>Put simply, you shouldn't HAVE to have an OP specified for spawn protection work.

Simple fix in case this is the solution you want to go with, Has been tested in a dev environment only.

I didn't add a configuration option because you can just disable the spawn protection anyway by setting spawn-protection in server.properties below 0.

At this stage it just comes down to whatever avoids the most confusion, having a default spawn protection, or not having it. Having it based on the number of OP's is just plain weird.

### CODE REVIEW

**Comment by Zidane** [comment]
@ryantheleach 

Thanks bud for this PR. Glad someone finally got to it!

@SpongePowered/developers Objections?

**Comment by me4502** [comment]
This is a much needed change

File: src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java [hunk 1]
```diff
@@ -55,4 +58,18 @@ public void setGuiEnabled() {
         this.guiIsEnabled = false;
     }
 
+    /**
+     * @author ryantheleach
+     *
+     * Purpose: This disables the new check in minecraft 1.8 that checks the OP list to check if there is no operators
+     * Reasoning: There has been complaints about 1.8 breaking spawn protection see: https://github.com/SpongePowered/SpongeCommon/issues/15
+     * UserList.hasEntries() is named incorrectly in MCP, it should be isEmpty
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java]: You might want to report this to [MCPBot-Issues](https://github.com/ModCoderPack/MCPBot-Issues/issues).

File: src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java [hunk 2]
```diff
@@ -55,4 +58,18 @@ public void setGuiEnabled() {
...
+     * The logic at time of writing is if OPList is empty, disable spawn protection. This has been replaced with if(false) disable spawn protection
+     */
+
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java]: Not sure if this newline is part of the style?

File: src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java (hunk 1 above)
Comment (ryantheleach) [inline src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java]: https://github.com/ModCoderPack/MCPBot-Issues/issues/27

**Comment by ryantheleach** [comment]
Build failing only on openjdk8 because travis couldn't find MCP

File: src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java (hunk 1 above)
Comment (Mumfrey) [inline src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java]: Or we could just change the mapping to `isEmpty`, is there any need to open an issue for such a minor thing?

**Comment by zml2008** [comment]
imo this ops list check should be replaced with a permissions check, which would match op levels as well, but also allow configuration with a permissions plugin.

File: src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java (hunk 1 above)
Comment (kenzierocks) [inline src/main/java/org/spongepowered/common/mixin/core/server/MixinDedicatedServer.java]: @mumfrey MCP changes that replace other ones need higher permissions, so we can't personally change it.

**Comment by zml2008** [comment]
#128 replaces this with a permission-controlled system that will block even if no ops are set.

**Pull request closed without merging**